exonum/src/events/noise/wrappers/sodium_wrapper/wrapper.rs

```diff
@@ -160,14 +162,19 @@ impl NoiseWrapper {
     // of AEAD authentication data. Therefore to calculate an actual message
     // length we need to subtract `TAG_LENGTH` multiplied by messages count
     // from `data.len()`.
-    fn decrypted_msg_len(&self, raw_message_len: usize) -> usize {
-        raw_message_len - TAG_LENGTH * (raw_message_len / MAX_MESSAGE_LENGTH)
+    fn decrypted_msg_len(raw_message_len: usize) -> usize {
+        raw_message_len - TAG_LENGTH * Self::tag_count(raw_message_len)
+    }
+
+    fn tag_count(message_len: usize) -> usize {
+        debug_assert!(message_len > 0);
+        1 + (message_len - 1) / MAX_MESSAGE_LENGTH
     }
 
     // In case of encryption we need to add `TAG_LENGTH` multiplied by messages count to
     // calculate actual message length.
-    fn encrypted_msg_len(&self, raw_message_len: usize) -> usize {
-        raw_message_len + TAG_LENGTH * ((raw_message_len / MAX_MESSAGE_LENGTH) + 1)
+    fn encrypted_msg_len(raw_message_len: usize) -> usize {
+        raw_message_len + TAG_LENGTH * Self::tag_count(raw_message_len)
```

Comment:
A minor observation on the code layout. I tend to put utility functions (i.e., the ones without **self** argument) outside the **impl** so that I don't have to prefix them with `Self::` which I personally find weird.